access/slackbot/cache.go

```diff
@@ -36,15 +36,15 @@ type Entry struct {
 // RequestCache holds pending request data.
 type RequestCache struct {
 	sync.Mutex
-	entries map[string]Entry
+	entries map[string]*Entry
```

Comment:
1. Since we only put and pop entries, there is no danger in sharing the pointer. Many caching solutions like `sync.Pool` has a recommendation to use pointers.
2. It causes lots of redundant allocations.
3. Well, yes, but here we allocate memory even when we missed the cache, which is no good.

> No need to touch this file anyway, eventually it will be completely removed when gravitational/teleport#3286 is done.

Can't disagree here :)

Reply:
> It causes lots of redundant allocations.

Stack allocations are a lot cheapier than heap allocations. And it's also expected to be safe as long as resulting structure is not very big. For example, look at what code is being generated here https://stackoverflow.com/questions/10866195/stack-vs-heap-allocation-of-structs-in-go-and-how-they-relate-to-garbage-collec

Reply:
I made some benchmarks and yep, it looks like there are no heap allocations, though I was pretty sure they should be.
Solution with pointers is a bit faster but it's not critical here.